wallet/src/Pos/Wallet/Web/Pending/Submission.hs

```diff
@@ -116,10 +118,15 @@ submitAndSavePtx PtxSubmissionHandlers{..} enqueue ptx@PendingTx{..} = do
     when walletTxCreationDisabled $
         throwM $ InternalError "Transaction creation is disabled by configuration!"
 
+    addOnlyNewPendingTx ptx
     ack <- submitTxRaw enqueue _ptxTxAux
     reportSubmitted ack
-    saveTx (_ptxTxId, _ptxTxAux) `catches` handlers ack
-    addOnlyNewPendingTx ptx
+    (saveTx (_ptxTxId, _ptxTxAux)
+        `catches` handlers ack)
+        `onException` creationFailedHandler
```

Comment:
Help me out to untangle this double layer of exception handling: are we really saying here: try to gracefully handle the exception (first `catches`) and if no other handler match, fail with a generic exception message (second `onException` block?)
  
If that's true, wouldn't the 2nd `Handler $ \e@SomeException{}` always match, making the second `onException` effectively unreachable?

Reply:
Good catch! The `creationFailedHandler` will never fire

Reply:
We should change this, to either remove the redundant handler, or make sure it can be reached.